### PR TITLE
Update precompiles registered addrs

### PR DESCRIPTION
**Pull request by AsceticBear**
See https://github.com/darwinia-network/darwinia/issues/901 and https://github.com/darwinia-network/darwinia/issues/906

1. Install dispatch precompile to the Pangoro network runtime.
2. Update `StateStorage` address to `1024`, `Dispatch` precompile address to `1025`. Leave others untouched and add some comments.

### CODE REVIEW

**Comment by AsceticBear** [comment]
@wuminzhe @hujw77 @xiaoch05 This change might be related to your work.  Pay attention to it.

File: node/runtime/pangolin/src/pallets/evm.rs [hunk 1]
```diff
@@ -146,22 +146,23 @@ where
...
 			a if a == addr(23) =>
 				Some(<EthereumBridge<R>>::execute(input, target_gas, context, is_static)),
 			a if a == addr(24) => Some(<Sub2SubBridge<
 				R,
 				ToPangoroMessageSender,
 				bm_pangoro::ToPangoroOutboundPayLoad,
 			>>::execute(input, target_gas, context, is_static)),
-			a if a == addr(25) =>
-				Some(<Dispatch<R>>::execute(input, target_gas, context, is_static)),
-			a if a == addr(27) => Some(<StateStorage<R, StorageFilter>>::execute(
```
Comment (HackFisher) [inline node/runtime/pangolin/src/pallets/evm.rs:162]: We can remove this in testnets, but for Crab, let's keep this id for a while as an legacy compatibility concern.

Util all apps remove use of deprecated ids, we can safely remove them then.

**Review by HackFisher — APPROVED** [review]

File: node/runtime/pangolin/src/pallets/evm.rs [hunk 2]
```diff
@@ -146,22 +146,23 @@ where
...
 				Some(<Transfer<R>>::execute(input, target_gas, context, is_static)),
+			// TODO: Delete EthereumBridge and Sub2SubBridge precompiles in the futures.
 			a if a == addr(23) =>
 				Some(<EthereumBridge<R>>::execute(input, target_gas, context, is_static)),
 			a if a == addr(24) => Some(<Sub2SubBridge<
 				R,
 				ToPangoroMessageSender,
 				bm_pangoro::ToPangoroOutboundPayLoad,
 			>>::execute(input, target_gas, context, is_static)),
-			a if a == addr(25) =>
```
Comment (HackFisher) [inline node/runtime/pangolin/src/pallets/evm.rs:160]: @xiaoch05 Are you using this precompile in helix on testnets?

And are you ok with removing it if you will need to re-deploy contracts or you would rather to keep this id too?

File: node/runtime/pangolin/src/pallets/evm.rs (hunk 1 above)
Comment (HackFisher) [reply node/runtime/pangolin/src/pallets/evm.rs:162]: Let's keep this id for dispatch precompile in pangolin too until Hexlix make their changes, and replacing with new precompiles id.

Comment (AsceticBear) [reply node/runtime/pangolin/src/pallets/evm.rs:162]: Okay.

Comment (xiaoch05) [reply node/runtime/pangolin/src/pallets/evm.rs:162]: remove this call on testnets can also cause failure of Helix ci.

File: node/runtime/pangolin/src/pallets/evm.rs (hunk 2 above)
Comment (xiaoch05) [reply node/runtime/pangolin/src/pallets/evm.rs:160]: ok, I will try to upgrade the contract on testnets after this update merged into pangolin network.

Comment (AsceticBear) [reply node/runtime/pangolin/src/pallets/evm.rs:160]: Please review again.

**Review by xiaoch05 — APPROVED** [review]

**Review by HackFisher — APPROVED** [review]

File: node/runtime/pangolin/src/pallets/evm.rs [hunk 3]
```diff
@@ -146,22 +146,28 @@ where
...
-				Some(<Dispatch<R>>::execute(input, target_gas, context, is_static)),
-			a if a == addr(27) => Some(<StateStorage<R, StorageFilter>>::execute(
+			// There are two StateStorage precompile instance now, the 25-StateStorage reserved to
+			// keep the compatibility, which will be removed in the future.
+			a if a == addr(25) => Some(<StateStorage<R, StorageFilter>>::execute(
```
Comment (HackFisher) [inline node/runtime/pangolin/src/pallets/evm.rs:163]: Hmm. here is a mistake?

Should be `Dispatch `

**Pull request merged**